addons/binding/org.openhab.binding.nest/src/main/java/org/openhab/binding/nest/internal/rest/NestStreamingRestClient.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * Copyright (c) 2010-2017 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.nest.internal.rest;
+
+import java.time.Duration;
+import java.util.List;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import javax.net.ssl.SSLContext;
+import javax.ws.rs.client.Client;
+import javax.ws.rs.client.ClientBuilder;
+
+import com.google.gson.GsonBuilder;
+import org.glassfish.jersey.SslConfigurator;
+import org.glassfish.jersey.media.sse.EventSource;
+import org.glassfish.jersey.media.sse.InboundEvent;
+import org.glassfish.jersey.media.sse.SseFeature;
+import org.openhab.binding.nest.internal.data.TopLevelData;
+import org.openhab.binding.nest.internal.data.TopLevelStreamingData;
+import org.openhab.binding.nest.internal.listener.NestStreamingDataListener;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.Gson;
+
+/**
+ * A client that generates events based on Nest streaming REST API Server-Sent Events (SSE).
+ *
+ * @author Wouter Born - Replace polling with REST streaming
+ */
+public class NestStreamingRestClient {
+
+    // Nest sends every 30 seconds a message to keep the connection alive
+    private static final long KEEP_ALIVE_MILLIS = Duration.ofSeconds(30).toMillis();
+
+    // Assume connection timeout when 2 keep alive message should have been received
+    private static final long CONNECTION_TIMEOUT_MILLIS = 2 * KEEP_ALIVE_MILLIS + KEEP_ALIVE_MILLIS / 2;
+
+    private static final String AUTH_REVOKED = "auth_revoked";
+    private static final String ERROR = "error";
+    private static final String KEEP_ALIVE = "keep-alive";
+    private static final String OPEN = "open";
+    private static final String PUT = "put";
+
+    private Logger logger = LoggerFactory.getLogger(NestStreamingRestClient.class);
```

Comment:
Maybe make logger `final` and put on the top but the last is just personal preference ;-)

Reply:
Added missing final modifier to loggers.